workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectCreation/GameObjectCreatorFromMetadata.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.IO;
+using Generated.Improbable;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectCreatorFromMetadata : IEntityGameObjectCreator
+    {
+        private readonly Dictionary<string, GameObject> cachedPrefabs
+            = new Dictionary<string, GameObject>();
+
+        private readonly string workerType;
+
+        public GameObjectCreatorFromMetadata(string workerType)
+        {
+            this.workerType = workerType;
+        }
+
+        public GameObject CreateGameObjectForEntityAdded(SpatialOSEntity entity, WorkerSystem worker)
+        {
+            if (!entity.HasComponent<Metadata.Component>())
+            {
+                return null;
+            }
+
+            var prefabName = entity.GetComponent<Metadata.Component>().EntityType;
+            if (!entity.HasComponent<Transform.Component>())
```

Comment:
Why do we need this? Is the idea that even though entities should be spawned with a default rotation, we still want to rely on our TransformSync features?

Reply:
that should be removed, we decided to use Position for the default one